gobblin-core/src/main/java/org/apache/gobblin/source/extractor/extract/restapi/RestApiConnector.java

```diff
@@ -192,8 +190,8 @@ private static boolean hasId(JsonObject json) {
         if (httpEntity != null) {
           EntityUtils.consume(httpEntity);
         }
-        if(httpResponse instanceof CloseableHttpResponse) {
-          this.closer.register((CloseableHttpResponse)httpResponse);
+        if(httpResponse instanceof Closeable) {
```

Comment:
since the `Closer` is our only resource management angle, should we log when not a `Closeable`, to give some trace before falling through the cracks?  better still would be to wrap within a holder that implements `Closeable`.

I did observe we still use the non-`Closeable`, `AutoRetryHttpClient` in some circumstances.  if that's what we happen to have would the wrapper use its `ClientConnectionManger` to manage resources?